### PR TITLE
add doc for smooth upgrade

### PR DESCRIPTION
**Pull request by ran-huang**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/14013
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: smooth-upgrade-tidb.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+The resumed DDL jobs are still executed in the order before the upgrade.
+
+## Limitations
+
+When using the smooth upgrade feature, note the following limitations.
+
+### Limitations on user operations
+
+* Before upgrading, if there is a canceling DDL job in the cluster, that is, an ongoing DDL job is canceled by the user, because the job in the canceling state cannot be paused, TiDB will retry. If the retry fails, an error is reported and the upgrade is exited.
```
[the remark this answers is not in the document]
Comment (qiancai) [reply smooth-upgrade-tidb.md:40]: Before the upgrade 和后面的 Before the upgrade 对应

File: smooth-upgrade-tidb.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+* During the upgrade, the following operations are not allowed:
+
+    * Run DDL operations on system tables (`mysql.*`, `information_schema.*`, `performance_schema.*`, `metrics_schema.*`).
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:44]:
```suggestion
    * Run DDL operations on system tables (`mysql.*`, `information_schema.*`, `performance_schema.*`, and `metrics_schema.*`).
```

File: smooth-upgrade-tidb.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    * Manually cancel, pause, or resume DDL jobs: `ADMIN CANCEL/PAUSE/RESUME DDL JOBS job_id [, job_id] ...;`.
+    * Import data.
+
+### Limitations on tools
+
+* During the upgrade, use of the following tools is not supported:
+
+    * BR: BR might replicate the paused DDL jobs to TiDB. The paused DDL jobs cannot be automatically resumed, which might cause the DDL jobs to be stuck later.
+
+    * DM and TiCDC: If you use DM or TiCDC to import SQL to TiDB during the upgrade process, and if one of the SQL statements contains DDL operations, the import operation is blocked and undefined errors might occur.
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:54]:
```suggestion
    * DM and TiCDC: If you use DM or TiCDC to import SQL statements to TiDB during the upgrade process, and if one of the SQL statements contains DDL operations, the import operation is blocked and undefined errors might occur.
```

File: smooth-upgrade-tidb.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+Starting from v7.1.0, when you upgrade TiDB to a higher version, TiDB supports smooth upgrade. This feature removes the limitations during the upgrade process and provides a more user-friendly upgrade experience. This feature is enabled by default and cannot be disabled.
+
+## Feature introduction
+
+Before the smooth upgrade feature is introduced, there are the following limitations on DDL operations during the upgrade process (see the *warning* section in [Upgrade TiDB Using TiUP](/upgrade-tidb-using-tiup.md#upgrade-tidb-using-tiup)):
+
+- Running DDL operations during the upgrade process might cause undefined behavior in TiDB.
+- Upgrading TiDB during the DDL operation might cause undefined behavior in TiDB.
+
+After the smooth upgrade feature is introduced, the upgrade process is no longer subject to the above limitations.
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:23]:
```suggestion
After the smooth upgrade feature is introduced, the upgrade process is no longer subject to the preceding limitations.
```

File: smooth-upgrade-tidb.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+>
+> Smooth upgrade is still an experimental feature.
+
+This document introduces the smooth upgrade feature of TiDB, which supports upgrading TiDB clusters without manually canceling DDL operations.
+
+Starting from v7.1.0, when you upgrade TiDB to a higher version, TiDB supports smooth upgrade. This feature removes the limitations during the upgrade process and provides a more user-friendly upgrade experience. This feature is enabled by default and cannot be disabled.
+
+## Feature introduction
+
+Before the smooth upgrade feature is introduced, there are the following limitations on DDL operations during the upgrade process (see the *warning* section in [Upgrade TiDB Using TiUP](/upgrade-tidb-using-tiup.md#upgrade-tidb-using-tiup)):
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:18]:
```suggestion
Before the smooth upgrade feature is introduced, there are the following limitations on DDL operations during the upgrade process (see the *Warning* content in [Upgrade TiDB Using TiUP](/upgrade-tidb-using-tiup.md#upgrade-tidb-using-tiup)):
```

File: smooth-upgrade-tidb.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+- Running DDL operations during the upgrade process might cause undefined behavior in TiDB.
+- Upgrading TiDB during the DDL operation might cause undefined behavior in TiDB.
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:21]:
```suggestion
- Upgrading TiDB during the DDL operations might cause undefined behavior in TiDB.
```

File: smooth-upgrade-tidb.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+After the smooth upgrade feature is introduced, the upgrade process is no longer subject to the above limitations.
+
+During the upgrade process, TiDB automatically performs the following operations without user intervention:
+
+1. Pause user DDL operations.
+2. Perform system DDL operations during the upgrade process.
```
Comment (qiancai) [inline smooth-upgrade-tidb.md:28]:
```suggestion
2. Perform system DDL operations for the upgrade.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: upgrade-tidb-using-tiup.md [hunk 8]
```diff
@@ -20,10 +20,11 @@ This document is targeted for the following upgrade paths:
...
+> 2. **DO NOT** run DDL statements during the upgrade process. Otherwise, the issue of undefined behavior might occur.
+> 3. **DO NOT** upgrade a TiDB cluster when a DDL statement is being executed in the cluster (usually for the time-consuming DDL statements such as `ADD INDEX` and the column type changes). Before the upgrade, it is recommended to use the [`ADMIN SHOW DDL`](/sql-statements/sql-statement-admin-show-ddl.md) command to check whether the TiDB cluster has an ongoing DDL job. If the cluster has a DDL job, to upgrade the cluster, wait until the DDL execution is finished or use the [`ADMIN CANCEL DDL`](/sql-statements/sql-statement-admin-cancel-ddl.md) command to cancel the DDL job before you upgrade the cluster.
+>
+> When upgrading from v7.1.0 to a later version, you can ignore the preceding warnings 2 and 3. For more information, see [TiDB Smooth Upgrade](/smooth-upgrade-tidb.md).
```
Comment (ran-huang) [inline upgrade-tidb-using-tiup.md:27]:
```suggestion
> If the TiDB version before upgrade is earlier than v7.1.0, you can ignore the preceding warnings 2 and 3. For more information, see [TiDB Smooth Upgrade](/smooth-upgrade-tidb.md).
```

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**